src/Games/NexusMods.Games.StardewValley/Installers/SMAPIInstaller.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        if (!installation.Is<StardewValley>()) return Common.Priority.None;
+
+        var installDataFiles = GetInstallDataFiles(files);
+        return installDataFiles.Length == 3
+            ? Common.Priority.Highest
+            : Common.Priority.None;
+    }
+
+    public ValueTask<IEnumerable<AModFile>> GetFilesToExtractAsync(GameInstallation installation,
+        Hash srcArchiveHash, EntityDictionary<RelativePath, AnalyzedFile> files,
+        CancellationToken ct = default)
+    {
+        return ValueTask.FromResult(GetFilesToExtract(installation, files));
+    }
+
+    private IEnumerable<AModFile> GetFilesToExtract(
+        GameInstallation installation,
+        EntityDictionary<RelativePath, AnalyzedFile> files)
+    {
+        var installDataFiles = GetInstallDataFiles(files);
+        if (installDataFiles.Length != 3) throw new UnreachableException($"{nameof(SMAPIInstaller)} should guarantee with {nameof(Priority)} that {nameof(GetInstallDataFiles)} returns 3 files when called from {nameof(GetFilesToExtractAsync)} but it has {installDataFiles.Length} files instead!");
+
+        KeyValuePair<RelativePath, AnalyzedFile> installDataFile;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            installDataFile = installDataFiles.First(kv => kv.Key.Parent.FileName.Equals("linux"));
+        } else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            installDataFile = installDataFiles.First(kv => kv.Key.Parent.FileName.Equals("macOS"));
+        } else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            installDataFile = installDataFiles.First(kv => kv.Key.Parent.FileName.Equals("windows"));
+        }
+        else
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        var (path, file) = installDataFile;
+        if (file is not AnalyzedArchive archive)
+            throw new UnreachableException($"{nameof(AnalyzedFile)} that has the file type {nameof(FileType.ZIP)} is not a {nameof(AnalyzedArchive)}");
+
+        var gameFolderPath = installation.Locations
+            .First(x => x.Key == GameFolderType.Game).Value;
+
+        var archiveContents = archive.Contents;
+
+        // TODO: install.dat is an archive inside an archive see https://github.com/Nexus-Mods/NexusMods.App/issues/244
+        // the basicFiles have to be extracted from the nested archive and put inside the game folder
+        // https://github.com/Pathoschild/SMAPI/blob/9763bc7484e29cbc9e7f37c61121d794e6720e75/src/SMAPI.Installer/InteractiveInstaller.cs#L380-L384
+        var basicFiles = archiveContents
+            .Where(kv => !kv.Key.Equals("unix-launcher.sh")).ToArray();
+
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // TODO: Replace game launcher (StardewValley) with unix-launcher.sh by overwriting the game file
```

Comment:
This also needs to happen for Gamepass, right now we instruct users to do it manually but I kinda wish it was part of the installer.

Reply:
You mean [this](https://stardewvalleywiki.com/Modding:Installing_SMAPI_on_Windows#Xbox_app)?

> 1. rename `Stardew Valley.exe` to another name such as `Stardew Valley original.exe`
> 2. make a copy of `StardewModdingAPI.exe` and name the copy `Stardew Valley.exe`